File: Computer-Cluster-Java-master/Regneklynge.java
import java.util.ArrayList;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class Regneklynge {
  private ArrayList<Rack> rackliste = new ArrayList<Rack>();
  private int nodePerRack;

  public Regneklynge(String filnavn) {

    File fil = new File(filnavn);

    try {
      Scanner sc = new Scanner(fil);
      nodePerRack = sc.nextInt();
      sc.nextLine();

      int antNoder = 0;
      int minne = 0;
      int proAnt = 0;

      while (sc.hasNextLine()) {

        String[] linje = sc.nextLine().split(" ");
        //linje = linje.split(" ");
        antNoder = Integer.parseInt(linje[0]);
        minne = Integer.parseInt(linje[1]);
        proAnt = Integer.parseInt(linje[2]);

        for (int i = 0; i < antNoder; i++) {
          Node n = new Node(minne, proAnt);
          leggTil(n);
        }
      }
    }

    catch (FileNotFoundException e) {
      System.out.println("File not found.");
    }
  }


  public void leggTil(Node node) {
    for (Rack r : rackliste) {
      if (!r.erFull()) {
        r.leggTil(node);
        return;
      }
    }

    //hvis det ikke finnes et ledig rack
    Rack r = new Rack(nodePerRack);
    r.leggTil(node);
    rackliste.add(r);
  }

  public int antProsessorer() {
    int antPro = 0;
    for (Rack r : rackliste) {
      antPro += r.antProsessorer();
    }

    return antPro;
  }

  public int noderMedNokMinne(int paakrevdMinne) {
    int nokMinne = 0;

    for (Rack r : rackliste) {
      nokMinne += r.nokMinne(paakrevdMinne);
    }

    return nokMinne;

  }


  public int hentAntRack() {
    return rackliste.size();
  }


}

File: Computer-Cluster-Python-master/regneklynge.py
#referer til ett eller flere Rack-objekter
from rack import Rack
from node import Node

class Regneklynge:
    #tar inn antall noder per rack i konstruktøren
    def __init__(self, filnavn):
        self._filnavn = filnavn
        self._racker = []
        self._nodePerRack = 0
        self.lesFraFil(self._filnavn)

    def lesFraFil(self, filnavn):
        f = open(filnavn)
        self._nodePerRack = int(f.readline())
        for linje in f:
            linje = linje.split(" ")
            for i in range(int(linje[0])):
                node = Node(int(linje[1]), int(linje[2]))
                self.leggTilNode(node)


    def leggTilNode(self, node):
        self._hentTomPlass().leggTilNode(node)

    #går gjennom Rackene
    #returnerer en tom plass hvis det finnes, hvis ikke lages et nytt Rack
    def _hentTomPlass(self):
        for rack in self._racker:
            if rack.erLedig():
                return rack
        #hvis det ikke finnes et ledig rack i racker-listen:
        nyrack = Rack(self._nodePerRack)
        self._racker.append(nyrack)
        return nyrack

    #returnerer antall prosessorer i regneklyngen
    def antProsessorer(self):
        ant = 0
        for rack in self._racker:
            ant += rack.hentProsessorantall()
        return ant

    def antRack(self):
        return len(self._racker)

    def noderMedNokMinne(self, paakrevdMinne):
        ant = 0
        for rack in self._racker:
            ant += rack.noderMedNokMinne(paakrevdMinne)
        print("Noder med minst " + str(paakrevdMinne) + "GB: " + str(ant))
        return ant

    def printInfo(self):
        racknr = 1
        for rack in self._racker:
            print("***** Rack " + str(racknr) + " *****")
            rack.printInfo()
            print()
            racknr += 1
